File: src/main/java/net/yakclient/mixin/internal/methodadapter/core/CoreBRPatternMatcher.java
package net.yakclient.mixin.internal.methodadapter.core;

import net.yakclient.mixin.internal.bytecode.ByteCodeUtils;
import net.yakclient.mixin.internal.instruction.Instruction;
import org.objectweb.asm.MethodVisitor;

public class CoreBRPatternMatcher extends CoreMixinPatternMatcher {
    public CoreBRPatternMatcher(MethodVisitor visitor, Instruction instructions) {
        super(visitor, instructions);
    }

    @Override
    public void visitInsn(int opcode) {
        if (ByteCodeUtils.isReturn(opcode)) this.executeInsn();
        super.visitInsn(opcode);
    }

    @Override
    public void visitInsn() {
    }
}

File: mixin-base/src/main/kotlin/net/yakclient/mixin/base/internal/methodadapter/core/CoreBRPatternMatcher.kt
package net.yakclient.mixin.base.internal.methodadapter.core

import net.yakclient.mixin.base.internal.bytecode.ByteCodeUtils.isReturn
import net.yakclient.mixin.base.internal.instruction.Instruction
import org.objectweb.asm.MethodVisitor

class CoreBRPatternMatcher(
    visitor: MethodVisitor,
    instructions: Instruction
) : CoreMixinPatternMatcher(visitor, instructions) {

    override fun visitInsn(opcode: Int) {
        if (isReturn(opcode)) this.executeInsn()
        super.visitInsn(opcode)
    }

    override fun visitInsn() {}
}